xmtp/Cargo.toml

```diff
@@ -10,13 +10,16 @@ path = "src/bin/update-schema.rs"
 
 # See more keys and their definitions at https://doc.rust-lang.org/cargo/reference/manifest.html
 
+[features]
+default = ["types"]
```

Comment:
> It allows xmtp_networking to pull in and extend the XmtpApiClient trait without pulling in the rest of xmtp.

I'm not sure I follow this. My understanding is:

1) XmtpApiClient is located inside of `./types.rs`
2) `xmtp/lib.rs` conditionally defines `pub mod types;` if the feature is enabled
3) `networking/Cargo.toml` has a dependency `xmtp = { path = "../xmtp", features = ["types"] }`

I don't immediately see how the `types` feature is being used to only access `XmtpApiClient` while not importing `xmtp` 

From what I can see, the feature definition in 3) looks redundant, as the feature is enabled by default. The code included in the compilation unit would be additive `Base + cfg(types).` rather than just `cfg(types)` in isolation. 

Further more compiling `xmtp` without the `types` feature results in compilation failures as `types.rs` includes Error and Address which are not protected behind feature gates. 

if the goal is to not pull in the rest of xmtp, my understanding was that the XmtpApi implementation would need to be in its own library crate as crates are the smallest unit of compilation